src/main/java/org/jenkinsci/plugins/github_branch_source/GitHubBuildStatusNotification.java

```diff
@@ -127,6 +131,31 @@ private static void createBuildCommitStatus(Run<?, ?> build, TaskListener listen
         }
     }
 
+    private static GitHubEnvAction getGitHubEnvAction(SCMRevision revision) {
```

Comment:
I would suggest moving this logic to the environment contributor

Reply:
I don't have access to the revision there as far as I know.
It's possible to look it up, but that requires an API call from memory, and  this plugin is very focused on minimising API calls as it's easy to go over the hourly limit